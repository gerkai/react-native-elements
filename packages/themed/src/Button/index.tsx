import { withTheme } from '../config';
import {
  Button,
  ButtonProps,
} from '@react-native-elements/base/dist/Button/Button';

export { Button };
export type { ButtonProps };
export default withTheme(Button, 'Button');
