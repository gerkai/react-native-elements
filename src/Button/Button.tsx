import React, { useCallback, useEffect } from 'react';
import {
  View,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  ActivityIndicator,
  Platform,
  StyleSheet,
  TouchableOpacityProps,
  TouchableNativeFeedbackProps,
  StyleProp,
  ViewStyle,
  ActivityIndicatorProps,
  TextStyle,
} from 'react-native';
import Color from 'color';
import { renderNode, color, RneFunctionComponent } from '../helpers';
import Icon, { IconNode } from '../Icon';
import { Theme } from '../config/theme';
import { TextProps } from '../Text';

const defaultLoadingProps = (
  type: 'solid' | 'clear' | 'outline',
  theme: Theme<ButtonProps> | undefined
): ActivityIndicatorProps => ({
  color: type === 'solid' ? 'white' : theme?.colors?.primary,
  size: 'small',
});

export type ButtonProps = TouchableOpacityProps &
  TouchableNativeFeedbackProps & {
    /** Add button title. */
    title?: string | React.ReactElement<{}>;

    /** Add additional styling for title component. */
    titleStyle?: StyleProp<TextStyle>;

    /** Add additional props for Text component. */
    titleProps?: TextProps;

    /** Add additional styling for button component. */
    buttonStyle?: StyleProp<ViewStyle>;

    /** Type of button. */
    type?: 'solid' | 'clear' | 'outline';

    /** Prop to display a loading spinner. */
    loading?: boolean;

    /** Add additional styling for loading component. */
    loadingStyle?: StyleProp<ViewStyle>;

    /** Add additional props for ActivityIndicator component. */
    loadingProps?: ActivityIndicatorProps;

    /** Styling for Component container. */
    containerStyle?: StyleProp<ViewStyle>;

    /** Displays a centered icon (when no title) or to the left (with text). (can be used along with iconRight as well). Can be an object or a custom component. */
    icon?: IconNode;

    /** Styling for Icon Component container. */
    iconContainerStyle?: StyleProp<ViewStyle>;

    /** Displays Icon to the right of title. Needs to be used along with `icon` prop. */
    iconRight?: boolean;

    /** Displays a linear gradient. See [usage](#lineargradient-usage). */
    linearGradientProps?: object;

    /** Component for user interaction. */
    TouchableComponent?: typeof React.Component;

    /** Component for container. */
    ViewComponent?: typeof React.Component;

    /** Disables user interaction. */
    disabled?: boolean;

    /** Style of the button when disabled. */
    disabledStyle?: StyleProp<ViewStyle>;

    /** Style of the title when disabled. */
    disabledTitleStyle?: StyleProp<TextStyle>;

    /** Add raised button styling (optional). Has no effect if `type="clear"`. */
    raised?: boolean;

    /** Displays Icon to the position mentioned. Needs to be used along with `icon` prop. */
    iconPosition?: 'left' | 'right' | 'top' | 'bottom';
  };

/** Buttons are touchable elements used to interact with the screen and to perform and operation.
 * They may display text, icons, or both. Buttons can be styled with several props to look a specific way.
 * Also receives all [TouchableNativeFeedback](http://reactnative.dev/docs/touchablenativefeedback.html#props) (Android) or [TouchableOpacity](http://reactnative.dev/docs/touchableopacity.html#props) (iOS) props. */
export const Button: RneFunctionComponent<ButtonProps> = ({
  TouchableComponent,
  containerStyle,
  onPress = () => console.log('Please attach a method to this component'),
  buttonStyle,
  type = 'solid',
  loading = false,
  loadingStyle,
  loadingProps: passedLoadingProps,
  title = '',
  titleProps,
  titleStyle: passedTitleStyle,
  icon,
  iconContainerStyle,
  iconRight = false,
  disabled = false,
  disabledStyle,
  disabledTitleStyle,
  raised = false,
  linearGradientProps,
  ViewComponent = View,
  theme,
  iconPosition = 'left',
  ...attributes
}) => {
  useEffect(() => {
    if (linearGradientProps && !ViewComponent) {
      console.error(
        "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
      );
    }
  });

  const handleOnPress = useCallback(
    (evt) => {
      if (!loading && !disabled) {
        onPress(evt);
      }
    },
    [loading, onPress, disabled]
  );

  // Refactor to Pressable
  const TouchableComponentInternal =
    TouchableComponent ||
    Platform.select({
      android: linearGradientProps ? TouchableOpacity : TouchableNativeFeedback,
      default: TouchableOpacity,
    });

  const titleStyle: StyleProp<TextStyle> = StyleSheet.flatten([
    {
      color: type === 'solid' ? 'white' : theme?.colors?.primary,
    },
    styles.title,
    passedTitleStyle,
    disabled && {
      color: color(theme?.colors?.disabled).darken(0.3).string(),
    },
    disabled && disabledTitleStyle,
  ]);

  const background =
    Platform.OS === 'android' && Platform.Version >= 21
      ? TouchableNativeFeedback.Ripple(
          Color(titleStyle?.color?.toString()).alpha(0.32).rgb().string(),
          true
        )
      : undefined;

  const loadingProps: ActivityIndicatorProps = {
    ...defaultLoadingProps(type, theme),
    ...passedLoadingProps,
  };

  const accessibilityState = {
    disabled: !!disabled,
    busy: !!loading,
  };
  const positionStyle = {
    top: 'column',
    bottom: 'column-reverse',
    left: 'row',
    right: 'row-reverse',
  };

  return (
    <View
      style={[
        styles.container,
        {
          borderRadius: 3 || styles.container.borderRadius,
        },
        containerStyle,
        raised && !disabled && type !== 'clear' && styles.raised,
      ]}
      testID="RNE_BUTTON_WRAPPER"
    >
      <TouchableComponentInternal
        onPress={handleOnPress}
        delayPressIn={0}
        activeOpacity={0.3}
        accessibilityRole="button"
        accessibilityState={accessibilityState}
        disabled={disabled}
        background={background}
        {...attributes}
      >
        <ViewComponent
          {...linearGradientProps}
          style={StyleSheet.flatten([
            styles.button,
            styles.buttonOrientation,
            {
              flexDirection:
                positionStyle[iconRight ? 'right' : iconPosition] || 'row',
            },
            {
              backgroundColor:
                type === 'solid' ? theme?.colors?.primary : 'transparent',
              borderColor: theme?.colors?.primary,
              borderWidth: type === 'outline' ? StyleSheet.hairlineWidth : 0,
            },
            buttonStyle,
            disabled &&
              type === 'solid' && {
                backgroundColor: theme?.colors?.disabled,
              },
            disabled &&
              type === 'outline' && {
                borderColor: color(theme?.colors?.disabled)
                  .darken(0.3)
                  .string(),
              },
            disabled && disabledStyle,
          ])}
        >
          {loading && (
            <ActivityIndicator
              style={StyleSheet.flatten([styles.loading, loadingStyle])}
              color={loadingProps.color}
              size={loadingProps.size}
              {...loadingProps}
            />
          )}
          {!loading &&
            icon &&
            renderNode(Icon, icon, {
              containerStyle: StyleSheet.flatten([
                styles.iconContainer,
                iconContainerStyle,
              ]),
            })}

          {!loading &&
            !!title &&
            renderNode(Text, title, {
              style: titleStyle,
              ...titleProps,
            })}
        </ViewComponent>
      </TouchableComponentInternal>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    padding: 8,
  },
  buttonOrientation: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    padding: 8,
  },
  container: {
    overflow: 'hidden',
    borderRadius: 3,
  },
  title: {
    fontSize: 16,
    textAlign: 'center',
    paddingVertical: 1,
    ...Platform.select({
      android: {
        fontFamily: 'sans-serif-medium',
      },
      default: {
        fontSize: 18,
      },
    }),
  },
  iconContainer: {
    marginHorizontal: 5,
  },
  raised: {
    backgroundColor: '#fff',
    overflow: 'visible',
    ...Platform.select({
      android: {
        elevation: 4,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  loading: {
    marginVertical: 2,
  },
});

Button.displayName = 'Button';
