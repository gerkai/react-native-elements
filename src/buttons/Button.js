import PropTypes from 'prop-types';
import React, { Component } from 'react';

import {
  StyleSheet,
  View,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  ActivityIndicator,
  Platform,
} from 'react-native';

import ViewPropTypes from '../config/ViewPropTypes';

class Button extends Component {
  log() {
    console.log('Please attach a method to this component');
  }

  render() {
    const {
      containerStyle,
      onPress,
      buttonStyle,
      clear,
      loading,
      loadingStyle,
      loadingProps,
      text,
      textStyle,
      textProps,
      icon,
      iconContainerStyle,
      iconRight,
      ...attributes
    } = this.props;

    // this is what RN Button does by default
    // https://github.com/facebook/react-native/blob/master/Libraries/Components/Button.js#L118
    const Touchable =
      Platform.OS === 'android' ? TouchableNativeFeedback : TouchableOpacity;

    return (
      <View style={[styles.container, containerStyle]}>
        <Touchable
          onPress={onPress || this.log.bind(this)}
          underlayColor={clear && 'transparent'}
          activeOpacity={clear && 0}
          style={{
            borderRadius:
              (buttonStyle &&
                buttonStyle.borderRadius &&
                buttonStyle.borderRadius) ||
              3,
          }}
          {...attributes}
        >
          <View
            style={[
              styles.button,
              clear && { backgroundColor: 'transparent', elevation: 0 },
              buttonStyle,
            ]}
          >
            {loading &&
              <ActivityIndicator
                animating={true}
                style={[styles.loading, loadingStyle]}
                color={(loadingProps && loadingProps.color) || 'white'}
                size={(loadingProps && loadingProps.size) || 'small'}
                {...loadingProps}
              />}
            {!loading &&
              icon &&
              !iconRight &&
              <View style={[styles.iconContainer, iconContainerStyle]}>
                {icon}
              </View>}
            {!loading &&
              <Text style={[styles.text, textStyle]} {...textProps}>
                {text || 'Welcome to\nReact Native Elements'}
              </Text>}
            {!loading &&
              icon &&
              iconRight &&
              <View style={[styles.iconContainer, iconContainerStyle]}>
                {icon}
              </View>}
          </View>
        </Touchable>
      </View>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string,
  textStyle: Text.propTypes.style,
  textProps: PropTypes.object,

  buttonStyle: ViewPropTypes.style,

  clear: PropTypes.bool,

  loading: PropTypes.bool,
  loadingStyle: ViewPropTypes.style,
  loadingProps: PropTypes.object,

  onPress: PropTypes.any,
  containerStyle: ViewPropTypes.style,

  icon: PropTypes.object,
  iconContainerStyle: ViewPropTypes.style,
  iconRight: PropTypes.bool,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
  },
  button: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    ...Platform.select({
      ios: {
        // iOS blue from https://developer.apple.com/ios/human-interface-guidelines/visual-design/color/
        backgroundColor: '#007AFF',
      },
      android: {
        elevation: 4,
        // Material design blue from https://material.google.com/style/color.html#color-color-palette
        backgroundColor: '#2196F3',
        borderRadius: 2,
      },
    }),
  },
  text: {
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
    padding: 8,
    ...Platform.select({
      ios: {
        fontSize: 18,
      },
      android: {
        fontWeight: '500',
      },
    }),
  },
  iconContainer: {
    marginHorizontal: 5,
  },
});

export default Button;
