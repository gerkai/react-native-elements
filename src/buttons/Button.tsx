import React from 'react';
import {
  View,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  ActivityIndicator,
  Platform,
  StyleSheet,
  TouchableOpacityProps,
  TouchableNativeFeedbackProps,
  StyleProp,
  ViewStyle,
  ActivityIndicatorProps,
  TextStyle,
} from 'react-native';
import Color from 'color';
import { withTheme } from '../config';
import { renderNode, color } from '../helpers';
import Icon, { IconNode } from '../icons/Icon';
import { Theme } from '../config/theme';
import { TextProps } from '../text/Text';

const defaultLoadingProps = (
  type: 'solid' | 'clear' | 'outline',
  theme: Theme
): ActivityIndicatorProps => ({
  color: type === 'solid' ? 'white' : theme.colors.primary,
  size: 'small',
});

export type ButtonProps = TouchableOpacityProps &
  TouchableNativeFeedbackProps & {
    title?: string | React.ReactElement<{}>;
    titleStyle?: StyleProp<TextStyle>;
    titleProps?: TextProps;
    buttonStyle?: StyleProp<ViewStyle>;
    type?: 'solid' | 'clear' | 'outline';
    loading?: boolean;
    loadingStyle?: StyleProp<ViewStyle>;
    loadingProps?: ActivityIndicatorProps;
    containerStyle?: StyleProp<ViewStyle>;
    icon?: IconNode;
    iconContainerStyle?: StyleProp<ViewStyle>;
    iconRight?: boolean;
    linearGradientProps?: object;
    TouchableComponent?: typeof React.Component;
    ViewComponent?: typeof React.Component;
    disabled?: boolean;
    disabledStyle?: StyleProp<ViewStyle>;
    disabledTitleStyle?: StyleProp<TextStyle>;
    raised?: boolean;
    theme?: Theme;
  };

const Button: React.FunctionComponent<ButtonProps> = (props: ButtonProps) => {
  if (props.linearGradientProps && !props.ViewComponent) {
    console.error(
      "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
    );
  }

  const handleOnPress = (evt) => {
    const {
      loading,
      onPress = () => console.log('Please attach a method to this component'),
    } = props;
    if (!loading) {
      onPress(evt);
    }
  };

  const {
    TouchableComponent,
    containerStyle,
    onPress,
    buttonStyle,
    type = 'solid',
    loading = false,
    loadingStyle,
    loadingProps: passedLoadingProps,
    title = '',
    titleProps,
    titleStyle: passedTitleStyle,
    icon,
    iconContainerStyle,
    iconRight = false,
    disabled = false,
    disabledStyle,
    disabledTitleStyle,
    raised = false,
    linearGradientProps,
    ViewComponent = View,
    theme,
    ...attributes
  } = props;

  // Refactor to Pressable
  const TouchableComponentInternal =
    TouchableComponent ||
    Platform.select({
      android: linearGradientProps ? TouchableOpacity : TouchableNativeFeedback,
      default: TouchableOpacity,
    });

  const titleStyle: StyleProp<TextStyle> = StyleSheet.flatten([
    { color: type === 'solid' ? 'white' : theme.colors.primary },
    styles.title,
    passedTitleStyle,
    disabled && { color: color(theme.colors.disabled).darken(0.3).string() },
    disabled && disabledTitleStyle,
  ]);

  const background =
    Platform.OS === 'android' && Platform.Version >= 21
      ? TouchableNativeFeedback.Ripple(
          Color(titleStyle.color.toString()).alpha(0.32).rgb().string(),
          true
        )
      : undefined;

  const loadingProps: ActivityIndicatorProps = {
    ...defaultLoadingProps(type, theme),
    ...passedLoadingProps,
  };

  const accessibilityState = {
    disabled: !!disabled,
    busy: !!loading,
  };

  return (
    <View
      style={[
        styles.container,
        {
          borderRadius: 3 || styles.container.borderRadius,
        },
        containerStyle,
        raised && !disabled && type !== 'clear' && styles.raised,
      ]}
    >
      <TouchableComponentInternal
        onPress={handleOnPress}
        delayPressIn={0}
        activeOpacity={0.3}
        accessibilityRole="button"
        accessibilityState={accessibilityState}
        disabled={disabled}
        background={background}
        {...attributes}
      >
        <ViewComponent
          {...linearGradientProps}
          style={StyleSheet.flatten([
            styles.button,
            {
              backgroundColor:
                type === 'solid' ? theme.colors.primary : 'transparent',
              borderColor: theme.colors.primary,
              borderWidth: type === 'outline' ? StyleSheet.hairlineWidth : 0,
            },
            buttonStyle,
            disabled &&
              type === 'solid' && { backgroundColor: theme.colors.disabled },
            disabled &&
              type === 'outline' && {
                borderColor: color(theme.colors.disabled).darken(0.3).string(),
              },
            disabled && disabledStyle,
          ])}
        >
          {loading && (
            <ActivityIndicator
              style={StyleSheet.flatten([styles.loading, loadingStyle])}
              color={loadingProps.color}
              size={loadingProps.size}
              {...loadingProps}
            />
          )}

          {!loading &&
            icon &&
            !iconRight &&
            renderNode(Icon, icon, {
              containerStyle: StyleSheet.flatten([
                styles.iconContainer,
                iconContainerStyle,
              ]),
            })}

          {!loading &&
            !!title &&
            renderNode(Text, title, {
              style: titleStyle,
              ...titleProps,
            })}

          {!loading &&
            icon &&
            iconRight &&
            renderNode(Icon, icon, {
              containerStyle: StyleSheet.flatten([
                styles.iconContainer,
                iconContainerStyle,
              ]),
            })}
        </ViewComponent>
      </TouchableComponentInternal>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    padding: 8,
  },
  container: {
    overflow: 'hidden',
    borderRadius: 3,
  },
  title: {
    fontSize: 16,
    textAlign: 'center',
    paddingVertical: 1,
    ...Platform.select({
      android: {
        fontFamily: 'sans-serif-medium',
      },
      default: {
        fontSize: 18,
      },
    }),
  },
  iconContainer: {
    marginHorizontal: 5,
  },
  raised: {
    backgroundColor: '#fff',
    overflow: 'visible',
    ...Platform.select({
      android: {
        elevation: 4,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  loading: {
    marginVertical: 2,
  },
});

export { Button };

export default withTheme<ButtonProps, {}>(Button, 'Button');
