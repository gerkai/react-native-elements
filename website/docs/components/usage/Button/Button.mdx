import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import Snack from './snack/index.md';

<Snack />
