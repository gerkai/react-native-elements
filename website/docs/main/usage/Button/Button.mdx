import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';

### Button by types

<Tabs
defaultValue="solid"
values={[
{ label: 'Solid', value: 'solid' },
{ label: 'Outline', value: 'outline' },
{ label: 'Clear', value: 'clear' },
]}
>

<TabItem value="solid">

```jsx live
<Button title="Solid Button" />
```

  </TabItem>
  <TabItem value="outline">

```jsx live
<Button title="Outline Button" type="outline" />
```

  </TabItem>
  <TabItem value="clear">

```jsx live
<Button title="Clear Button" type="clear" />
```

  </TabItem>
</Tabs>

### Button with icon object

```jsx live
<Button
  icon={{
    name: 'add',
    size: 20,
    color: 'white',
  }}
  title="Button with icon object"
/>
```

### Button with right icon

Add `iconRight` to show the icon on the right side of the button.

```jsx live
<Button
  icon={{
    name: 'add',
    size: 20,
    color: 'white',
  }}
  title="Button with right icon"
  iconRight
/>
```

### Button with icon object

You can use external `Icon` components inside the button.
Example:

```jsx live
<Button
  icon={<FontAwesomeIcon name="arrow-right" size={16} color="white" />}
  title="Button with icon component"
  iconRight
  titleStyle={{ marginHorizontal: 8 }}
/>
```

### Button with loading

For handling button loading add `loading` or `loading={true}` to the button. Also you can combine `loading` and `disabled`
props to make the button unclickable.

```jsx live
<Button title="Loading button" buttonStyle={{ width: 150 }} loading disabled />
```
