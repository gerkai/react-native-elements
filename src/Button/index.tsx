import { withTheme } from '../config';
import { Button, ButtonProps } from './Button';

export { Button };
export type { ButtonProps };
export default withTheme(Button, 'Button');
