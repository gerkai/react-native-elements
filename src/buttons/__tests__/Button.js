/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../Button';

describe('Button Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Button />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should be call onPress events', () => {
    const onPress = jest.fn();
    console.log = jest.fn();
    const wrapper = shallow(<Button />);

    // Call default onPress
    wrapper
      .find('TouchableOpacity')
      .props()
      .onPress();

    expect(console.log.mock.calls[0][0]).toBe(
      `Please attach a method to this component`
    );

    wrapper.setProps({ onPress });

    // Call our custom onPress
    wrapper
      .find('TouchableOpacity')
      .props()
      .onPress();

    expect(onPress).toBeCalled();
  });

  it('should have ripple on android version 21 and higher', () => {
    jest.mock('Platform', () => ({
      OS: 'android',
      Version: 25,
    }));

    const wrapper = shallow(<Button />);
    expect(wrapper.length).toBe(1);
    jest.resetModules();
  });

  it('should have normal ripple on android version 20 and lower', () => {
    jest.mock('Platform', () => ({
      OS: 'android',
      Version: 20,
    }));

    const wrapper = shallow(<Button />);
    expect(wrapper.length).toBe(1);
    jest.resetModules();
  });

  it('should warn the user when using linearGradient without it installed', () => {
    console.error = jest.fn();
    shallow(
      <Button
        linearGradientProps={{ colors: ['#4c669f', '#3b5998', '#192f6a'] }}
      />
    );

    expect(console.error.mock.calls[0][0]).toBe(
      `You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}`
    );
  });
});
