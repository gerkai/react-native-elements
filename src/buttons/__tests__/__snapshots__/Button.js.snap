// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    underlayColor="transparent"
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <customIconComponent
        color="white"
        iconRight={true}
        name="wifi"
        size={22}
        style={
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "fontSize": 20,
            },
          ]
        }
      />
      <TextElement
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontSize": 17.5,
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`Button Component should render with default icon 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onPress={[Function]}
    underlayColor="transparent"
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        iconRight={true}
        name="wifi"
        size={22}
        style={
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "fontSize": 20,
            },
          ]
        }
      />
      <TextElement
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontSize": 17.5,
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`Button Component should render with icon type 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onPress={[Function]}
    underlayColor="transparent"
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#007AFF",
            "borderRadius": 3,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontSize": 17.5,
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`Button Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onPress={[Function]}
    underlayColor="transparent"
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <TextElement
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontSize": 17.5,
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Welcome to
        React Native Elements
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
